### PR TITLE
AP_RTC: fix oldest_acceptable_date to be in micros

### PR DESCRIPTION
**Pull request by Jaaaky**
Also update date to 2022-01-01 00:00 UTC

### CODE REVIEW

**Review by nkruzan — COMMENTED** [review]
Costs us small amount of flash, can you please explain the benefit.

**Comment by Jaaaky** [comment]
> Costs us small amount of flash, can you please explain the benefit.

I just fixed it. A feature was not working as expected and I fixed it.
The benefit would be to avoid very bad time setting by crazy or spoofed GPS for example.

File: libraries/AP_RTC/AP_RTC.cpp [hunk 1]
```diff
@@ -41,7 +41,7 @@ const AP_Param::GroupInfo AP_RTC::var_info[] = {
 
 void AP_RTC::set_utc_usec(uint64_t time_utc_usec, source_type type)
 {
-    const uint64_t oldest_acceptable_date = 1546300800000; // 2019-01-01 0:00
+    const uint64_t oldest_acceptable_date = 1640995200ULL*1000*1000; // 2022-01-01 0:00 in microseconds
```
Comment (peterbarker) [inline libraries/AP_RTC/AP_RTC.cpp:44]:
```suggestion
    const uint64_t oldest_acceptable_date_us = 1640995200ULL*1000*1000; // 2022-01-01 0:00
```

... should've been there from the start :-/

**Review by peterbarker — CHANGES_REQUESTED** [review]
Good catch.

File: libraries/AP_RTC/AP_RTC.cpp (hunk 1 above)
Comment (Jaaaky) [reply libraries/AP_RTC/AP_RTC.cpp]: Done

Comment (peterbarker) [reply libraries/AP_RTC/AP_RTC.cpp]: @Jaaaky sorry, I think you missed the bit about adding `us` to the variable name :-)

I've taken the liberty of making that change here.

**Review by peterbarker — APPROVED** [review]

**Comment by peterbarker** [comment]
Merged, thanks @Jaaaky

**Pull request merged**